test/test_queue_monkeypatch.py

```diff
@@ -0,0 +1,29 @@
+from __future__ import absolute_import
+
+import unittest
+
+import urllib3
+from urllib3.exceptions import EmptyPoolError
+import Queue
+
+class BadError(Exception):
+    """
+    This should not be raised.
+    """
+    pass
+
+Queue.Empty = BadError
+
+
+class TestConnectionPool(unittest.TestCase):
```

Comment:
I think we could have a better name for this test case here: `class TestMonkeypatchResistance` maybe?